Separate ideas into languages/other

MichalMarsalek opened the pull request:
Most of the ideas shown at `/ideas` are suggestions to add new languages. This means other ideas are harder to notice. This PR aims at solving this by separating the ideas into two groups: "Languages related" and "Other".

Please note that I have written no lines in Go before this.

JRaspass commented on views/ideas.html:
```diff
@@ -17,7 +17,18 @@ <h2>{{ .Name }}</h2>
+    <h1 class=span>Languages</h1>
+
+{{ range .Data.Languages }}
+    <a class="card color" href="//github.com/code-golf/code-golf/issues/{{ .ID }}" title="{{ .Title }}">
```
We ought to be able to avoid duplicating the identical markup with block/define? https://pkg.go.dev/text/template#hdr-Actions

MichalMarsalek replied:
Great! I've also never seen that templating system so I didn't bother trying to find out if/how it can be done. :)

JRaspass commented on views/ideas.html:
```diff
@@ -17,7 +17,18 @@ <h2>{{ .Name }}</h2>
+        {{ .ThumbsDown }}
+    </a>
+{{ end }}
+    <h1 class=span>Other ideas</h1>
```
title case please, Other Ideas, also maybe it should go first? there's generally fewer other ideas

MichalMarsalek replied:
Yeah, I thought about putting Other Ideas first, but it didn't really make sense to me to call it "Other" then.

JRaspass commented on routes/ideas.go:
```diff
@@ -15,8 +16,9 @@ func Ideas(w http.ResponseWriter, r *http.Request) {
-		Holes []*config.Hole
-		Ideas []idea
+		Holes     []*config.Hole
+		Languages []idea
```
combine these, `Ideas, Languages []idea`

MichalMarsalek replied:
Done.

JRaspass reviewed (changes requested):
Awesome idea, just a couple of suggestions.

MichalMarsalek reviewed (commented):
Hmm, does something like
```go
{{define "IdeasTable"}}
  {{ range . }}
    <a class="card color" href="//github.com/code-golf/code-golf/issues/{{ .ID }}" title="{{ .Title }}">
        <h2 class=span>{{ .Title }}</h2>
        <svg><use href="#thumbs-up"/></svg>
        {{ .ThumbsUp }}
        <svg><use href="#thumbs-down"/></svg>
        {{ .ThumbsDown }}
    </a>
  {{end}}
{{end}}

    <h1 class=span>Ideas</h1>
    {{template "IdeasTable" .Data.Ideas}}

    <h1 class=span>Language Suggestions</h1>
    {{template "IdeasTable" .Data.Languages}}
```
work instead?

JRaspass commented:
> Hmm, does something like
> 
> ```go
> {{define "IdeasTable"}}
>   {{ range . }}
>     <a class="card color" href="//github.com/code-golf/code-golf/issues/{{ .ID }}" title="{{ .Title }}">
>         <h2 class=span>{{ .Title }}</h2>
>         <svg><use href="#thumbs-up"/></svg>
>         {{ .ThumbsUp }}
>         <svg><use href="#thumbs-down"/></svg>
>         {{ .ThumbsDown }}
>     </a>
>   {{end}}
> {{end}}
> 
>     <h1 class=span>Ideas</h1>
>     {{template "IdeasTable" .Data.Ideas}}
> 
>     <h1 class=span>Language Suggestions</h1>
>     {{template "IdeasTable" .Data.Languages}}
> ```
> 
> work instead?

Yeah I like that, might as well fold the range into the define as it's common to both. Minor nit: could you add a single space around some of those template tags, define, end, etc.

MichalMarsalek commented:
> 

I added the spaces. I'm not sure whether you overlooked I moved the range into the define in my comment (I didn't commit it because I didn't know if it was the proper way) or whether you are talking about something else. Can you elaborate?

JRaspass commented:
> > Yeah I like that, might as well fold the range into the define as it's common to both. Minor nit: could you add a single space around some of those template tags, define, end, etc.
> 
> I added the spaces. I'm not sure whether you overlooked I moved the range into the define in my comment (I didn't commit it because I didn't know if it was the proper way) or whether you are talking about something else. Can you elaborate?

Yeah I was basing my comments on the current state of the PR, I liked the code sample you showed in this chat, with the `{{ range . }}`, the PR as it stands has the range ops outside the define for me - https://github.com/code-golf/code-golf/pull/587/files

MichalMarsalek commented:
> > > Yeah I like that, might as well fold the range into the define as it's common to both. Minor nit: could you add a single space around some of those template tags, define, end, etc.
> > 
> > 
> > I added the spaces. I'm not sure whether you overlooked I moved the range into the define in my comment (I didn't commit it because I didn't know if it was the proper way) or whether you are talking about something else. Can you elaborate?
> 
> Yeah I was basing my comments on the current state of the PR, I liked the code sample you showed in this chat, with the `{{ range . }}`, the PR as it stands has the range ops outside the define for me - https://github.com/code-golf/code-golf/pull/587/files

Ah, my bad! I read your comment as an independent recommendation to do that, rather than as an approval.

MichalMarsalek commented:
Thank you so much for helping a Go noob make this PR mergeable!
